packages/create-sitecore-jss/src/templates/nextjs/src/components/SafeHydrate.tsx

```diff
@@ -0,0 +1,14 @@
+﻿import dynamic from 'next/dynamic';
+import React from 'react';
+
+type props = {
+  children: React.ReactNode;
+};
+
+// This is a component when imported dynamically will not have any server side rendering.
+// This is a workaround to prevent hydration errors when using JSS components.
```

Comment:
This is only necessary in Sitecore editors, should we reflect that in the comment? e.g.
```suggestion
// This is used as a workaround to prevent hydration errors when using SSR in Sitecore editors.
```